platform/iphone/view_controller.mm

```diff
@@ -117,12 +117,109 @@ - (BOOL)prefersStatusBarHidden {
 	return YES;
 }
 
-#ifdef GAME_CENTER_ENABLED
+#pragma mark - Touch Handling via UIControl
+- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {
+	UITouch *touch = [touches anyObject];
+
+	[self.activeTouches addObject:touch];
+
+	CGPoint touchPoint = [self scaledPoint:[touch locationInView:self.view]];
+	OSIPhone::get_singleton()->touch_press([self.activeTouches indexOfObject:touch], touchPoint.x, touchPoint.y, true, touch.tapCount > 1);
+}
+
+- (void)touchesMoved:(NSSet *)touches withEvent:(UIEvent *)event {
+	UITouch *touch = [touches anyObject];
+
+	CGPoint touchPoint = [self scaledPoint:[touch locationInView:self.view]];
+	CGPoint previousPoint = [self scaledPoint:[touch previousLocationInView:self.view]];
+	OSIPhone::get_singleton()->touch_drag([self.activeTouches indexOfObject:touch], previousPoint.x, previousPoint.y, touchPoint.x, touchPoint.y);
+}
+
+- (void)touchesEnded:(NSSet *)touches withEvent:(UIEvent *)event {
+	UITouch *touch = [touches anyObject];
+
+	CGPoint touchPoint = [self scaledPoint:[touch locationInView:self.view]];
+	OSIPhone::get_singleton()->touch_press([self.activeTouches indexOfObject:touch], touchPoint.x, touchPoint.y, false, false);
+
+	[self.activeTouches removeObject:touch];
+}
+
+- (void)touchesCancelled:(NSSet *)touches withEvent:(UIEvent *)event {
+	UITouch *touch = [touches anyObject];
+
+	OSIPhone::get_singleton()->touches_cancelled();
+
+	[self.activeTouches removeObject:touch];
+}
+
+- (CGPoint)scaledPoint:(CGPoint)point {
+	return CGPointMake(point.x * self.view.contentScaleFactor, point.y * self.view.contentScaleFactor);
+}
+
+#pragma mark - Keyboard Input
+- (BOOL)canBecomeFirstResponder {
+	return YES;
```

Comment:
We must return `YES` from this method to allow this object to become first in the `UIResponder` chain. This means all inputs will be sent to this class first to determine if they are actionable or not. We're returning `YES` so we can later become the first responder and present the keyboard.